src/SegmentedString.chpl

```diff
@@ -1524,7 +1536,35 @@ module SegmentedString {
     The SegFunction called by computeOnSegments for isdigit
   */
   inline proc stringIsDigit(ref values, rng) throws {
-    return interpretAsString(values, rng, borrow=true).isDigit();
+    use In1d;
+    const special_digits = "⁰¹²³⁴⁵⁶⁷⁸⁹₀₁₂₃₄₅₆₇₈₉";
+    const myString = interpretAsString(values, rng, borrow=true);
+    // test if string is all regular digits for early out
+    if myString.isDigit(){
+      return true;
+    }
+    // test if string is all alphanumeric characters for early out
+    else if myString.isAlpha() || myString.isEmpty() {
+      return false;
+    }
+    // string contains at least one special digit character, full test
+    else {
+      proc toUint64(c) {
+        var ret:uint = 0;
+        var i = 0;
+        for b in c.bytes() {
+          var tmp = b:uint;
+          ret |= (tmp << (i*8));
+          i += 1;
+        }
+        return ret;
+      }
```

Comment:
prob worth adding a comment to explain why this function is here/ what it's doing. I don't think it's obvious that it's here to handle multi-byte cases (or how it does that) and to make it easier to use our integer In1d proc

Reply:
Added